crates/ide/src/diagnostics.rs

```diff
@@ -66,12 +73,8 @@ pub(crate) fn diagnostics(
             res.borrow_mut().push(diagnostic_with_fix(d, &sema));
         })
         // Only collect experimental diagnostics when they're enabled.
-        .filter(|diag| !diag.is_experimental() || enable_experimental);
-
-    if let Some(disabled_diagnostics) = disabled_diagnostics {
-        // Do not collect disabled diagnostics.
-        sink_builder = sink_builder.filter(move |diag| !disabled_diagnostics.contains(diag.name()));
-    }
+        .filter(|diag| !(diag.is_experimental() && config.disable_experimental))
+        .filter(|diag| !config.disabled.contains(diag.name()));
```

Comment:
If I understand things correctly, this will add an additional function call for each diagnostics run, even when there are no disabled diagnostics (and as long as the focus is to reduce amount of false-positives, I think we normally expect no disabled diagnostics). Is this not an issue?

Reply:
Initially this code looked like:

```rust
if !disabled_diagnostics.is_empty() {
    sink_builder = sink_builder.filter(|diag| !config.disabled.contains(diag.name()));
}
```

Reply:
Nothing is a performance issue until proven otherwise by a benchmark. This particular case is a cold-path -- there will be dozens of diagnostics at worst.

Reply:
Makes sense, thanks :)